packages/apps/tools/src/components/Global/ModuleExplorer/index.tsx

```diff
@@ -1,89 +1,88 @@
-  onModuleExpand: ISidePanelProps['onModuleExpand'];
+  onModuleClick: ISidePanelProps<IncompleteModuleModel>['onModuleClick'];
   onActiveModuleChange: IEditorProps['onActiveModuleChange'];
-  onTabClose: IEditorProps['onTabClose'];
+  // onTabClose: IEditorProps['onTabClose'];
+  items: ISidePanelProps<IncompleteModuleModel>['data'];
+  onReload: ISidePanelProps<IncompleteModuleModel | Outline>['onReload'];
+  onExpandCollapse: ISidePanelProps<
+    IncompleteModuleModel | Outline
+  >['onExpandCollapse'];
 }
 
 const ModuleExplorer = ({
-  modules,
-  openedModules: fetchedModules,
-  onModuleClick,
-  onInterfaceClick,
-  onInterfacesExpand,
-  onModuleExpand,
+  items,
+  onReload,
+  onExpandCollapse,
   onActiveModuleChange,
-  onTabClose,
-}: IModuleExplorerProps): React.JSX.Element => {
+  openedModules: _openedModules,
+}: IModuleExplorerProps) => {
   const [activeModule, setActiveModule] = useState<IChainModule>();
-  const results = getModulesMap(modules);
   const [openedModules, setOpenedModules] =
-    useState<IChainModule[]>(fetchedModules);
+    useState<IChainModule[]>(_openedModules);
+  let outlineItems: TreeItem<Outline>[] = [];
 
-  const updateOpenedModules = useCallback(
-    (result: IChainModule) => {
-      setOpenedModules((prev) => {
-        const alreadyOpened = prev.find((openedModule) => {
-          return (
-            openedModule.moduleName === result.moduleName &&
-            openedModule.chainId === result.chainId &&
-            openedModule.network === result.network
-          );
-        });
-
-        if (alreadyOpened) {
-          return prev;
-        }
-
-        let add = result;
-
-        const enhanced = modules.find((module) => {
-          return (
-            module.moduleName === result.moduleName &&
-            module.chainId === result.chainId
-          );
-        });
-
-        if (enhanced) {
-          add = enhanced;
-        }
-
-        return [...prev, add];
-      });
-    },
-    [modules],
-  );
+  if (activeModule) {
+    outlineItems = chainModuleToOutlineTreeItems(activeModule, items);
+  }
 
   return (
     <div className={containerStyle}>
-      <SidePanel
-        results={results}
-        onResultClick={(result) => {
-          updateOpenedModules(result);
-          onModuleClick(result);
-          setActiveModule(result);
-        }}
-        onInterfaceClick={(result) => {
-          updateOpenedModules(result);
-          onInterfaceClick(result);
-          setActiveModule(result);
+      <SidePanel<IncompleteModuleModel | Outline>
+        data={[
+          {
+            children: items,
+            title: 'Explorer',
+            key: 'explorer',
+            label: 'Explorer',
+            data: 'explorer',
+          },
+          {
+            title: 'Outline',
+            key: 'outline',
+            label: 'Outline',
+            children: outlineItems,
+            data: 'outline',
+          },
+        ]}
```

Comment:
I see some duplication.. Cant we just use the label prop and fill in the others with a capitalize function or lowercase function for the other parts?